djstripe/contrib/rest_framework/serializers.py

```diff
@@ -8,27 +8,108 @@
 """
 
 from rest_framework import serializers
-from rest_framework.serializers import ModelSerializer
+from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.serializers import ModelSerializer, ValidationError
 
-from djstripe.models import Subscription
+from djstripe.enums import SubscriptionStatus
+from djstripe.models import Customer, Plan, SetupIntent, Subscription
+from djstripe.settings import CANCELLATION_AT_PERIOD_END
 
+from .mixins import AutoCustomerModelSerializerMixin
 
-class SubscriptionSerializer(ModelSerializer):
-    """A serializer used for the Subscription model."""
 
-    class Meta:
-        """Model class options."""
+class SubscriptionSerializer(AutoCustomerModelSerializerMixin, ModelSerializer):
+    """A base serializer used for the Subscription model."""
 
+    class Meta:
         model = Subscription
         exclude = ["default_tax_rates"]
 
+    plan = serializers.PrimaryKeyRelatedField(
```

Comment:
This exposes the internal id of the plan (ie the numeric row id), I think it'd be better to use the  stripe id field (eg "plan_123") instead?

I think this should be possible using `SlugRelatedField(slug_field="id")`